src/js/components/Chart/calcs.js

```diff
@@ -90,74 +114,82 @@ export const calcBounds = (values, options = {}) => {
       maxY = alignMax(maxY, intervalY);
     }
 
-    if (minY < 0 && maxY > 0 && Math.abs(minY) !== Math.abs(maxY)) {
-      // Adjust min and max when crossing 0 to ensure 0 will be shown on
-      // the Y axis based on the number of steps.
-      if (stepsY === 1) {
-        const largest = Math.max(Math.abs(minY), Math.abs(maxY));
-        minY = -largest;
-        maxY = largest;
-      } else {
-        let stepInterval = (maxY - minY) / stepsY;
-        const minSteps = minY / stepInterval;
-        const maxSteps = maxY / stepInterval;
-        if (Math.abs(minSteps) < Math.abs(maxSteps)) {
-          // more above than below
-          stepInterval = maxY / Math.floor(maxSteps);
-          maxY = stepInterval * Math.floor(maxSteps);
-          minY = stepInterval * Math.floor(minSteps);
-        } else {
-          // more below than above
-          stepInterval = Math.abs(minY / Math.ceil(minSteps));
-          minY = stepInterval * Math.ceil(minSteps);
-          maxY = stepInterval * Math.ceil(maxSteps);
-        }
-      }
-    }
+    if (options.direction === 'vertical')
+      [minX, maxX] = adjustToShowZero(minX, maxX, stepsX);
+    else [minY, maxY] = adjustToShowZero(minY, maxY, stepsY);
+
+    // if options.direction is present, the results are delivered in { x, y }
+    // object structure. If options.direction is not present, the results are
+    // delivered in [x, y] array structure, for backwards compatibility
+    result = options.direction
+      ? { x: { min: minX, max: maxX }, y: { min: minY, max: maxY } }
+      : [
+          [minX, maxX],
+          [minY, maxY],
+        ];
+  } else {
+    result = options.direction ? { x: {}, y: {} } : [[], []];
   }
 
-  let bounds;
-  if (calcValues.length)
-    bounds = [
-      [minX, maxX],
-      [minY, maxY],
-    ];
-  else bounds = [[], []];
-  return bounds;
+  return result;
 };
 
+// if options.direction is present, the results are delivered in { x, y }
+// object structure. If options.direction is not present, the results are
+// delivered in [x, y] array structure, for backwards compatibility
 export const calcs = (values = [], options = {}) => {
+  const vertical = options.direction === 'vertical';
+
   // the number of steps is one less than the number of labels
-  const steps = options.steps || [1, 1];
-  const [stepsX, stepsY] = steps;
+  const { x: stepsX, y: stepsY } = normalizeSteps(options.steps);
+
+  // bounds is { x: { min, max }, y: { min, max } } when options.direction is
+  // present and [[min, max], [min, max]] if not, for backwards compatibility
```

Comment:
Is this returned to the caller? And that is why we have to maintain this structure?

Reply:
The structure doesn't seem to align with the docs -- unless I'm misreading? https://v2.grommet.io/datachart#bounds